From c486acb9c71eda3103ed443bbdacd39a8c57be28 Mon Sep 17 00:00:00 2001
From: Randy Li <randy.li@rock-chips.com>
Date: Mon, 17 Apr 2017 10:25:49 +0800
Subject: [PATCH] video: rockchip: rkvdec: add a new device id

RKVDEC second generation uses a new device id.
It is the new generation of the RKV decoder found on
the RK3328 platform.

Change-Id: I63891b7f774e68d8820f1a9c88052795af37f99d
Signed-off-by: Randy Li <randy.li@rock-chips.com>
---
 drivers/video/rockchip/vcodec/vcodec_hw_info.h | 1 +
 drivers/video/rockchip/vcodec/vcodec_service.c | 8 +++++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/vcodec/vcodec_hw_info.h b/drivers/video/rockchip/vcodec/vcodec_hw_info.h
index c9102c6beae0..a394650a653f 100644
--- a/drivers/video/rockchip/vcodec/vcodec_hw_info.h
+++ b/drivers/video/rockchip/vcodec/vcodec_hw_info.h
@@ -28,6 +28,7 @@ enum VPU_HW_ID {
 	VPU_ID_4831		= 0x4831,
 	HEVC_ID			= 0x6867,
 	RKV_DEC_ID		= 0x6876,
+	RKV_DEC_ID2             = 0x3410,
 	VPU2_ID			= 0x0000,
 };
 
diff --git a/drivers/video/rockchip/vcodec/vcodec_service.c b/drivers/video/rockchip/vcodec/vcodec_service.c
index 5fba743e63b4..f8b6760e1de9 100644
--- a/drivers/video/rockchip/vcodec/vcodec_service.c
+++ b/drivers/video/rockchip/vcodec/vcodec_service.c
@@ -175,6 +175,12 @@ static const struct vcodec_info vcodec_info_set[] = {
 		.task_info      = task_vpu2,
 		.trans_info     = trans_vpu2,
 	},
+	{
+		.hw_id		= RKV_DEC_ID2,
+		.hw_info	= &hw_rkvdec,
+		.task_info	= task_rkv,
+		.trans_info	= trans_rkv,
+	},
 };
 
 /* Both VPU1 and VPU2 */
@@ -2411,7 +2417,7 @@ static int vcodec_subdev_probe(struct platform_device *pdev,
 	}
 
 	data->child_dev = device_create(data->cls, dev,
-		data->dev_t, "%s", name);
+		data->dev_t, NULL, "%s", name);
 
 	platform_set_drvdata(pdev, data);
 
-- 
2.35.3

